### PR TITLE
Add Metric statistic table for instance, Volume, NoSQL

### PR DESCRIPTION
**Pull request by ParthaI**
**_Tables Added_**
- `oci_core_boot_volume_metric_read_ops`
- `oci_core_boot_volume_metric_read_ops_daily`
- `oci_core_boot_volume_metric_read_ops_hourly`
- `oci_core_boot_volume_metric_write_ops`
- `oci_core_boot_volume_metric_write_ops_daily`
- `oci_core_boot_volume_metric_write_ops_hourly`
- `oci_core_instance_metric_cpu_utilization`
- `oci_core_instance_metric_cpu_utilization_daily`
- `oci_core_instance_metric_cpu_utilization_hourly`
- `oci_nosql_table_metric_read_throttle_count`
- `oci_nosql_table_metric_read_throttle_count_daily`
- `oci_nosql_table_metric_read_throttle_count_hourly`
- `oci_nosql_table_metric_storage_utilization`
- `oci_nosql_table_metric_storage_utilization_daily`
- `oci_nosql_table_metric_storage_utilization_hourly`
- `oci_nosql_table_metric_write_throttle_count`
- `oci_nosql_table_metric_write_throttle_count_daily`
- `oci_nosql_table_metric_write_throttle_count_hourly`

# Integration test logs
<details>
  <summary>Logs</summary>
  
```
N/A
```
</details>

# Example query results
<details>
  <summary>Results</summary>

# Table: oci_core_instance_metric_cpu_utilization
_____  

### Basic Info
```sql
select
  instance_id,
  timestamp,
  minimum,
  maximum,
  average,
  sample_count
from
  oci_core_instance_metric_cpu_utilization
order by
  instance_id,
  timestamp;
```

```
+-------------------------------------------------------------------------------------+---------------------+----------------------+--------------------+---------------------+--------------+
| instance_id                                                                         | timestamp           | minimum              | maximum            | average             | sample_count |
+-------------------------------------------------------------------------------------+---------------------+----------------------+--------------------+---------------------+--------------+
| ocid1.instance.oc1.iad.anuwcljt6igdexaca4mgk62f3onzaa6i2xfi6aoayx2rccrolnri4ar6sz2q | 2021-07-11 06:40:00 | 0                    | 2.1489255372318388 | 0.4351259779941456  | 30           |
| ocid1.instance.oc1.iad.anuwcljt6igdexaca4mgk62f3onzaa6i2xfi6aoayx2rccrolnri4ar6sz2q | 2021-07-11 06:45:00 | 0                    | 1.499999999996362  | 0.39179840587701686 | 30           |
| ocid1.instance.oc1.iad.anuwcljt6igdexaca4mgk62f3onzaa6i2xfi6aoayx2rccrolnri4ar6sz2q | 2021-07-11 06:50:00 | 0                    | 0.9500000000025466 | 0.3501918651268563  | 30           |
| ocid1.instance.oc1.iad.anuwcljt6igdexaca4mgk62f3onzaa6i2xfi6aoayx2rccrolnri4ar6sz2q | 2021-07-11 06:55:00 | 0                    | 1.049475262373054  | 0.3618193413843329  | 30           |
| ocid1.instance.oc1.iad.anuwcljt6igdexaca4mgk62f3onzaa6i2xfi6aoayx2rccrolnri4ar6sz2q | 2021-07-11 07:00:00 | 0                    | 4.095904095955552  | 0.47671942954100566 | 30           |
| ocid1.instance.oc1.iad.anuwcljt6igdexaca4mgk62f3onzaa6i2xfi6aoayx2rccrolnri4ar6sz2q | 2021-07-11 07:05:00 | 0                    | 2.049999999975407  | 0.3851476497095862  | 30           |
| ocid1.instance.oc1.iad.anuwcljt6igdexaca4mgk62f3onzaa6i2xfi6aoayx2rccrolnri4ar6sz2q | 2021-07-11 07:10:00 | 0                    | 2.1510755377462885 | 0.43183093381480414 | 30           |
| ocid1.instance.oc1.iad.anuwcljt6igdexaca4mgk62f3onzaa6i2xfi6aoayx2rccrolnri4ar6sz2q | 2021-07-11 07:15:00 | 0                    | 1.9028542814207876 | 0.4269094081994613  | 30           |
| ocid1.instance.oc1.iad.anuwcljt6igdexaca4mgk62f3onzaa6i2xfi6aoayx2rccrolnri4ar6sz2q | 2021-07-11 07:20:00 | 0                    | 0.8004002001090175 | 0.34184096218973214 | 30           |
| ocid1.instance.oc1.iad.anuwcljt6igdexaca4mgk62f3onzaa6i2xfi6aoayx2rccrolnri4ar6sz2q | 2021-07-11 07:25:00 | 0                    | 3.8499999999482677 | 0.44349513845866784 | 30           |
| ocid1.instance.oc1.iad.anuwcljt6igdexaca4mgk62f3onzaa6i2xfi6aoayx2rccrolnri4ar6sz2q | 2021-07-11 07:30:00 | 0                    | 0.9504752376218001 | 0.36182345591513393 | 30           |
| ocid1.instance.oc1.iad.anuwcljt6igdexaca4mgk62f3onzaa6i2xfi6aoayx2rccrolnri4ar6sz2q | 2021-07-11 07:35:00 | 0                    | 1.150575287642174  | 0.3785443347283556  | 30           |
| ocid1.instance.oc1.iad.anuwcljt6igdexaca4mgk62f3onzaa6i2xfi6aoayx2rccrolnri4ar6sz2q | 2021-07-11 07:40:00 | 0                    | 3.999999999996362  | 0.5701309463271205  | 30           |
| ocid1.instance.oc1.iad.anuwcljt6igdexaca4mgk62f3onzaa6i2xfi6aoayx2rccrolnri4ar6sz2q | 2021-07-11 07:45:00 | 0                    | 1.5507753876920216 | 0.3950059171874113  | 30           |
| ocid1.instance.oc1.iad.anuwcljt6igdexaca4mgk62f3onzaa6i2xfi6aoayx2rccrolnri4ar6sz2q | 2021-07-11 07:50:00 | 0                    | 1.0499999999956344 | 0.35346593550383976 | 30           |
| ocid1.instance.oc1.iad.anuwcljt6igdexaca4mgk62f3onzaa6i2xfi6aoayx2rccrolnri4ar6sz2q | 2021-07-11 07:55:00 | 0                    | 2.301150575293448  | 0.3801918126192595  | 30           |
| ocid1.instance.oc1.iad.anuwcljt6igdexaca4mgk62f3onzaa6i2xfi6aoayx2rccrolnri4ar6sz2q | 2021-07-11 08:00:00 | 0                    | 4.547726136981606  | 0.4917918859999322  | 30           |

```

### CPU Over 1% average

```sql
select
  instance_id,
  timestamp,
  round(minimum::numeric,2) as min_cpu,
  round(maximum::numeric,2) as max_cpu,
  round(average::numeric,2) as avg_cpu,
  sample_count
from
  aws_ec2_instance_metric_cpu_utilization
where average > 1
order by
  instance_id,
  timestamp;
```

```
+---------------------+---------------------+---------+---------+---------+--------------+
| instance_id         | timestamp           | min_cpu | max_cpu | avg_cpu | sample_count |
+---------------------+---------------------+---------+---------+---------+--------------+
| i-0c227068c46fc6521 | 2021-07-14 13:04:00 | 0.29    | 7.08    | 1.74    | 5            |
+---------------------+---------------------+---------+---------+---------+--------------+
```

# Table: oci_core_instance_metric_cpu_utilization_daily
_____  

### Basic Info

```sql
select
  instance_id,
  timestamp,
  minimum,
  maximum,
  average,
  sample_count
from
  oci_core_instance_metric_cpu_utilization_daily
order by
  instance_id,
  timestamp;
```

```
+-------------------------------------------------------------------------------------+---------------------+---------+--------------------+---------------------+--------------+
| instance_id                                                                         | timestamp           | minimum | maximum            | average             | sample_count |
+-------------------------------------------------------------------------------------+---------------------+---------+--------------------+---------------------+--------------+
| ocid1.instance.oc1.iad.anuwcljt6igdexaca4mgk62f3onzaa6i2xfi6aoayx2rccrolnri4ar6sz2q | 2021-06-30 07:00:00 | 0       | 93.46733668341723  | 0.49049575011366164 | 7537         |
| ocid1.instance.oc1.iad.anuwcljt6igdexaca4mgk62f3onzaa6i2xfi6aoayx2rccrolnri4ar6sz2q | 2021-07-01 07:00:00 | 0       | 6.4500000000197275 | 0.44943895609113765 | 8639         |
| ocid1.instance.oc1.iad.anuwcljt6igdexaca4mgk62f3onzaa6i2xfi6aoayx2rccrolnri4ar6sz2q | 2021-07-02 07:00:00 | 0       | 5.497251374312103  | 0.45241368471374355 | 8640         |
| ocid1.instance.oc1.iad.anuwcljt6igdexaca4mgk62f3onzaa6i2xfi6aoayx2rccrolnri4ar6sz2q | 2021-07-03 07:00:00 | 0       | 7.3109664496423035 | 0.4562670069981587  | 8640         |
| ocid1.instance.oc1.iad.anuwcljt6igdexaca4mgk62f3onzaa6i2xfi6aoayx2rccrolnri4ar6sz2q | 2021-07-04 07:00:00 | 0       | 6.606606606577367  | 0.46154008813381164 | 8640         |
| ocid1.instance.oc1.iad.anuwcljt6igdexaca4mgk62f3onzaa6i2xfi6aoayx2rccrolnri4ar6sz2q | 2021-07-05 07:00:00 | 0       | 5.302651325693635  | 0.45981103187194555 | 8640         |
| ocid1.instance.oc1.iad.anuwcljt6igdexaca4mgk62f3onzaa6i2xfi6aoayx2rccrolnri4ar6sz2q | 2021-07-06 07:00:00 | 0       | 5.349999999936263  | 0.4496522288864595  | 8639         |
| ocid1.instance.oc1.iad.anuwcljt6igdexaca4mgk62f3onzaa6i2xfi6aoayx2rccrolnri4ar6sz2q | 2021-07-07 07:00:00 | 0       | 5.555555555619284  | 0.45800615490651225 | 8640         |
| ocid1.instance.oc1.iad.anuwcljt6igdexaca4mgk62f3onzaa6i2xfi6aoayx2rccrolnri4ar6sz2q | 2021-07-08 07:00:00 | 0       | 6.653326663256907  | 0.45064106072076693 | 8559         |
| ocid1.instance.oc1.iad.anuwcljt6igdexaca4mgk62f3onzaa6i2xfi6aoayx2rccrolnri4ar6sz2q | 2021-07-09 07:00:00 | 0       | 5.647176411721766  | 0.4535577552949724  | 8640         |
| ocid1.instance.oc1.iad.anuwcljt6igdexaca4mgk62f3onzaa6i2xfi6aoayx2rccrolnri4ar6sz2q | 2021-07-10 07:00:00 | 0       | 52.450980385198406 | 0.4449793483194458  | 8634         |
| ocid1.instance.oc1.iad.anuwcljt6igdexaca4mgk62f3onzaa6i2xfi6aoayx2rccrolnri4ar6sz2q | 2021-07-11 07:00:00 | 0       | 6.121424987392195  | 0.4202349581596094  | 8640         |
| ocid1.instance.oc1.iad.anuwcljt6igdexaca4mgk62f3onzaa6i2xfi6aoayx2rccrolnri4ar6sz2q | 2021-07-12 07:00:00 | 0       | 6.0469765116756795 | 0.4312030595953487  | 8639         |
| ocid1.instance.oc1.iad.anuwcljt6igdexaca4mgk62f3onzaa6i2xfi6aoayx2rccrolnri4ar6sz2q | 2021-07-13 07:00:00 | 0       | 5.649999999873543  | 0.42879664804042833 | 8637         |
| ocid1.instance.oc1.iad.anuwcljt6igdexaca4mgk62f3onzaa6i2xfi6aoayx2rccrolnri4ar6sz2q | 2021-07-14 07:00:00 | 0       | 5.197401299300884  | 0.4302198127020741  | 8640         |
| ocid1.instance.oc1.iad.anuwcljt6igdexaca4mgk62f3onzaa6i2xfi6aoayx2rccrolnri4ar6sz2q | 2021-07-15 07:00:00 | 0       | 29.59081836252323  | 0.439624702329049   | 8637         |
| ocid1.instance.oc1.iad.anuwcljt6igdexaca4mgk62f3onzaa6i2xfi6aoayx2rccrolnri4ar6sz2q | 2021-07-16 07:00:00 | 0       | 5.597201399236617  | 0.43559196116795157 | 8640         |
+-------------------------------------------------------------------------------------+---------------------+---------+--------------------+---------------------+--------------+
```

### CPU Over 0.452% average

```sql
select
  instance_id,
  timestamp,
  round(minimum::numeric,2) as min_cpu,
  round(maximum::numeric,2) as max_cpu,
  round(average::numeric,2) as avg_cpu,
  sample_count
from
  oci_core_instance_metric_cpu_utilization_daily
where average > 0.452
order by
  instance_id,
  timestamp;
```

```
+-------------------------------------------------------------------------------------+---------------------+---------+---------+---------+--------------+
| instance_id                                                                         | timestamp           | min_cpu | max_cpu | avg_cpu | sample_count |
+-------------------------------------------------------------------------------------+---------------------+---------+---------+---------+--------------+
| ocid1.instance.oc1.iad.anuwcljt6igdexaca4mgk62f3onzaa6i2xfi6aoayx2rccrolnri4ar6sz2q | 2021-06-30 07:00:00 | 0.00    | 93.47   | 0.49    | 7537         |
| ocid1.instance.oc1.iad.anuwcljt6igdexaca4mgk62f3onzaa6i2xfi6aoayx2rccrolnri4ar6sz2q | 2021-07-02 07:00:00 | 0.00    | 5.50    | 0.45    | 8640         |
| ocid1.instance.oc1.iad.anuwcljt6igdexaca4mgk62f3onzaa6i2xfi6aoayx2rccrolnri4ar6sz2q | 2021-07-03 07:00:00 | 0.00    | 7.31    | 0.46    | 8640         |
| ocid1.instance.oc1.iad.anuwcljt6igdexaca4mgk62f3onzaa6i2xfi6aoayx2rccrolnri4ar6sz2q | 2021-07-04 07:00:00 | 0.00    | 6.61    | 0.46    | 8640         |
| ocid1.instance.oc1.iad.anuwcljt6igdexaca4mgk62f3onzaa6i2xfi6aoayx2rccrolnri4ar6sz2q | 2021-07-05 07:00:00 | 0.00    | 5.30    | 0.46    | 8640         |
| ocid1.instance.oc1.iad.anuwcljt6igdexaca4mgk62f3onzaa6i2xfi6aoayx2rccrolnri4ar6sz2q | 2021-07-07 07:00:00 | 0.00    | 5.56    | 0.46    | 8640         |
| ocid1.instance.oc1.iad.anuwcljt6igdexaca4mgk62f3onzaa6i2xfi6aoayx2rccrolnri4ar6sz2q | 2021-07-09 07:00:00 | 0.00    | 5.65    | 0.45    | 8640         |
+-------------------------------------------------------------------------------------+---------------------+---------+---------+---------+--------------+
```
</details>

### CODE REVIEW

File: docs/tables/oci_core_instance_metric_cpu_utilization.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Table: oci_core_instance_metric_cpu_utilization
+
+OCI Monitoring Metrics Explorer provide data about the performance of your systems.  The `oci_core_instance_metric_cpu_utilization` table provides metric statistics at 5 minute intervals for the most recent 5 days.
```
Comment (Subhajit97) [inline docs/tables/oci_core_instance_metric_cpu_utilization.md:3]:
```suggestion
OCI Monitoring Metrics Explorer provide data about the performance of your systems. The `oci_core_instance_metric_cpu_utilization` table provides metric statistics at 5 minute intervals for the most recent 5 days.
```

File: docs/tables/oci_core_instance_metric_cpu_utilization.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+
```
Comment (Subhajit97) [inline docs/tables/oci_core_instance_metric_cpu_utilization.md:5]:
```suggestion
```

File: docs/tables/oci_core_instance_metric_cpu_utilization_hourly.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
+q# Table: oci_core_instance_metric_cpu_utilization_hourly
```
Comment (Subhajit97) [inline docs/tables/oci_core_instance_metric_cpu_utilization_hourly.md:1]:
```suggestion
# Table: oci_core_instance_metric_cpu_utilization_hourly
```

File: docs/tables/oci_core_instance_metric_cpu_utilization_hourly.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+OCI Monitoring Metrics Explorer provide data about the performance of your systems.  The `oci_core_instance_metric_cpu_utilization_hourly` table provides metric statistics at 60 minute intervals for the most recent 60 days.
+
+
```
Comment (Subhajit97) [inline docs/tables/oci_core_instance_metric_cpu_utilization_hourly.md:5]:
```suggestion
OCI Monitoring metrics explorer provide data about the performance of your systems. The `oci_core_instance_metric_cpu_utilization_hourly` table provides metric statistics at 60 minute intervals for the most recent 60 days.

```

File: docs/tables/oci_nosql_table_metric_write_throttle_count.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  sample_count
+from
+  oci_nosql_table_metric_write_throttle_count
+order by
+  table_name,
+  timestamp;
+```
+
+### Intervals where write throttle count exceded 100 average
+```sql
```
Comment (Subhajit97) [inline docs/tables/oci_nosql_table_metric_write_throttle_count.md:27]:
```suggestion
### Intervals where write throttle count exceeded 100 average

```sql
```

File: docs/tables/oci_nosql_table_metric_write_throttle_count_daily.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+  sample_count
+from
+  oci_nosql_table_metric_write_throttle_count_daily
+order by
+  table_name,
+  timestamp;
+```
+
+### Intervals where write throttle count exceded 100 average
+```sql
```
Comment (Subhajit97) [inline docs/tables/oci_nosql_table_metric_write_throttle_count_daily.md:27]:
```suggestion
### Intervals where write throttle count exceeded 100 average

```sql
```

File: oci/monitoring_metric.go [hunk 7]
```diff
@@ -0,0 +1,279 @@
...
+// append the common metric columns onto the column list
+func MonitoringMetricColumns(columns []*plugin.Column) []*plugin.Column {
+	return append(columns, commonMonitoringMetricColumns()...)
+}
+
+func commonMonitoringMetricColumns() []*plugin.Column {
+	return []*plugin.Column{
+		{
+			Name:        "compartment_id",
+			Description: "The ID of the Compartment.",
```
Comment (Subhajit97) [inline oci/monitoring_metric.go:25]:
```suggestion
			Description: "The ID of the compartment.",
```

File: oci/monitoring_metric.go [hunk 8]
```diff
@@ -0,0 +1,279 @@
...
+}
+
+type MetricData struct {
+	CompartmentId *string
+	PointValue    *float64
+	Timestamp     *time.Time
+}
+
+func listMonitoringMetricStastics(ctx context.Context, d *plugin.QueryData, granularity string, namespace string, metricName string, dimensionName string, dimensionValue string, compartmentId string) (*monitoring.SummarizeMetricsDataResponse, error) {
+	plugin.Logger(ctx).Trace("listMonitoringMetricStastics ==>")
```
Comment (Subhajit97) [inline oci/monitoring_metric.go:154]:
```suggestion
	plugin.Logger(ctx).Trace("listMonitoringMetricStastics")
```

File: oci/monitoring_metric.go [hunk 9]
```diff
@@ -0,0 +1,279 @@
...
+
+	// Min statistics
+	metricDetails.Query = &querystringMin
+	requestParam.SummarizeMetricsDataDetails = metricDetails
+	minStatistics, err := session.MonitoringClient.SummarizeMetricsData(ctx, requestParam)
+	if err != nil {
+		return nil, err
+	}
+	metricDetailsMin := filterMetricStatistic(minStatistics)
+	fmt.Println(metricDetailsMin)
```
Comment (Subhajit97) [inline oci/monitoring_metric.go:202]: comment additional debugging logs

File: oci/plugin.go [hunk 10]
```diff
@@ -28,66 +28,77 @@ func Plugin(ctx context.Context) *plugin.Plugin {
...
+			"oci_cloud_guard_target":                             tableCloudGuardTarget(ctx),
+			"oci_core_boot_volume_backup":                        tableCoreBootVolumeBackup(ctx),
+			"oci_core_dhcp_options":                              tableCoreDhcpOptions(ctx),
+			"oci_core_drg":                                       tableCoreDrg(ctx),
+			"oci_core_image":                                     tableCoreImage(ctx),
+			"oci_core_instance":                                  tableCoreInstance(ctx),
+			"oci_core_internet_gateway":                          tableCoreInternetGateway(ctx),
+			"oci_core_instance_metric_cpu_utilization":           tableOciCoreInstanceMetricCpuUtilization(ctx),
+			"oci_core_instance_metric_cpu_utilization_hourly":    tableOciCoreInstanceMetricCpuUtilizationHourly(ctx),
+			"oci_core_instance_metric_cpu_utilization_daily":     tableOciCoreInstanceMetricCpuUtilizationDaily(ctx),
```
Comment (Subhajit97) [inline oci/plugin.go:48]: Sort the list

File: oci/table_oci_core_instance_metric_cpu_utilization.go [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+		List: &plugin.ListConfig{
+			ParentHydrate: listCoreInstances,
+			Hydrate:       listCoreInstanceMetricCpuUtilization,
+		},
+		GetMatrixItem: BuildCompartementRegionList,
+		Columns: MonitoringMetricColumns(
+			[]*plugin.Column{
+				{
+					Name:        "instance_id",
+					Description: "The Core Instance ID.",
```
Comment (Subhajit97) [inline oci/table_oci_core_instance_metric_cpu_utilization.go:26]:
```suggestion
					Name:        "id",
					Description: "The OCID of the instance.",
```

**Review by Subhajit97 — APPROVED** [review]

**Comment by Subhajit97** [comment]
Closes #190 #191 #199 #200 #201 #216 #217 #218 #219 #220 #221 #222 #225 #226 #227 #228 #229 #230

**Review by rajlearner17 — APPROVED** [review]
LGTM

**Pull request merged**